src/NativeScript/ObjC/Constructor/ObjCConstructorBase.mm

```diff
@@ -181,52 +181,48 @@ static void writeAdapter(ExecState* execState, const JSValue& value, void* buffe
     visitor.append(constructor->_initializers.begin(), constructor->_initializers.end());
 }
 
-static WTF::String computeInitializerJSName(ExecState* execState, JSFinalObject* initializer, MarkedArgumentBuffer& arguments) {
+static JSValue getInitializerForSwiftStyleConstruction(ExecState* execState, ObjCConstructorBase* constructor, JSFinalObject* initializer, MarkedArgumentBuffer& arguments) {
     PropertyNameArray properties(execState, PropertyNameMode::Strings);
     initializer->getOwnPropertyNames(initializer, execState, properties, EnumerationMode(DontEnumPropertiesMode::Exclude, JSObjectPropertiesMode::Include));
+    if (properties.size() == 0) {
+        return JSValue();
+    }
 
     VM& vm = execState->vm();
     WTF::StringBuilder builder;
     builder.reserveCapacity(32);
-    for (auto& property : properties) {
-        const WTF::String& propertyString = property.string();
-        const WTF::String firstCharacter = propertyString.substringSharingImpl(0, 1).upper();
-        const WTF::String rest = propertyString.substringSharingImpl(1);
-
-        builder.append(firstCharacter);
-        builder.append(rest);
+    for (size_t i = 0; i < properties.size(); i++) {
+        JSC::Identifier property = properties[i];
+        builder.append(property.string());
+        builder.append(':');
         arguments.append(initializer->getDirect(vm, property));
     }
 
-    return builder.toString();
-}
-
-static JSValue getInitializerForSwiftStyleConstruction(ExecState* execState, JSValue prototype, WTF::String jsName) {
-    static WTF::String initWith("initWith", WTF::String::ConstructFromLiteral);
-    static WTF::String init("init", WTF::String::ConstructFromLiteral);
-    static WTF::String error("Error", WTF::String::ConstructFromLiteral);
-
-    JSValue value;
-#define RETURN_IF_CELL  \
-    if (value.isCell()) \
-    return value
-
-    if (jsName.isEmpty()) {
-        return value;
+    const char* ctorMetadata = builder.toString().utf8().data();
```

Comment:
I think that `utf8()` returns a temporary `CString` that holds the `data()`. After this statement it should be destroyed, deleting its storage. Consider changing the `ctorMetadata` to be `CString`.